### PR TITLE
ci: publish docker image

### PR DESCRIPTION
**Pull request by Guldoman**
This should publish a container image on ghcr.
At the moment this only pushes the tag `latest`. Should we add proper versions?
I'm using https://www.npmjs.com/package/http-server as entrypoint; is that ok, or should I look at using something else (nginx)?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+FROM archlinux:base-devel AS build
```
Comment (takase1121) [inline Dockerfile]: Have you investigated using the emscripten docker image?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+WORKDIR /build
+
+COPY . .
+
+RUN rm -rf build-emscripten-wasm32 github-pages && \
+    source /etc/profile.d/emscripten.sh && \
+    bash scripts/gh-pages-deploy.sh
+
+FROM node:current-alpine
+
```
Comment (takase1121) [inline Dockerfile:8]: The nginx docker seems pretty easy to configure. Just copy the output to /usr/share/nginx/html and you're all set.

File: scripts/gh-pages-deploy.sh [hunk 3]
```diff
@@ -62,12 +62,3 @@ node shell/closure.js "${dest_dir}/lite-xl-files.min.js" "${dest_dir}/lite-xl-fi
 # these files can be removed for final distribution
 rm -rf "$(pwd)/${dest_dir}/lite-xl.js" "$(pwd)/${dest_dir}/lite-xl-files.js" \
         "${data_dir}" "$(pwd)/${dest_dir}/doc"
-
-package_name="github-pages"
-echo "Creating a compressed archive ${package_name}"
-
-tar \
```
Comment (takase1121) [inline scripts/gh-pages-deploy.sh:69]: Have you tested this to work with the GH pages publishing workflow? It doesn't seem to create the tar file for publishing.

File: Dockerfile (hunk 1 above)
Comment (Guldoman) [reply Dockerfile]: Not yet. Will look into it.

File: Dockerfile (hunk 2 above)
Comment (Guldoman) [reply Dockerfile:8]: I'll test it out.

File: scripts/gh-pages-deploy.sh (hunk 3 above)
Comment (Guldoman) [reply scripts/gh-pages-deploy.sh:69]: The tar is now created by https://github.com/lite-xl/playground/blob/0a56d8c7ca427cfb4e7c64592b7e9bc72b9f91b7/.github/workflows/deploy.yml#L72-L73

I haven't tried actually publishing, but the artifact that gets created is basically the same to the one this repo creates.
Example action result: https://github.com/Guldoman/playground/actions/runs/11261084291

**Comment by Guldoman** [comment]
This is now using `emscripten/emsdk` for building and `nginx` for serving.
Example of published page: https://guldoman.github.io/playground

It looks like `nginx` has a `50x.html` file in the standard location, should we delete it, or is it ok to keep?

**Comment by takase1121** [comment]
Keep it.

**Pull request merged**